modules/cpr/src/main/java/org/atmosphere/config/managed/ManagedAtmosphereHandler.java

```diff
@@ -167,7 +179,13 @@ public void onClose(AtmosphereResourceEvent event) {
 
             MethodInfo.EncoderObject e = message(resource, body);
             if (e != null && e.encodedObject != null) {
-                IOUtils.deliver(new Managed(e.encodedObject), null, e.methodInfo.deliverTo, resource);
+            	if ( e.methodInfo.deliverTo == DeliverTo.DELIVER_TO.RESOURCE) {
+					AtmosphereResource res = getHandshakeResource(resource.getRequest());
```

Comment:
Can you try this instead of the `getHandshakeResource`:
```java
AtmosphereResource res = atmosphereResourceFactory.find(resource.uuid());
```
That will saves us one lookup in the attribute's map. The uuid should be the proper one. If that works just update this patch. Thanks for all the help!